flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/expressions/ExpressionParser.scala

```diff
@@ -59,6 +59,8 @@ object ExpressionParser extends JavaTokenParsers with PackratParsers {
           Literal(str.toInt)
         } else if (str.endsWith("f") | str.endsWith("F")) {
           Literal(str.toFloat)
+        } else if (str.endsWith(".")) {
```

Comment:
Since the 'cast()' function parsing is accepting '.' as '.cast()' we need to do this way. Else the parser does not know that '.' is specific to that operation of 'cast'. But I am not so well versed with this parser so suggestions welcome.